### DISCUSSION TITLE
Color-code Pandas Dataframe dtypes in IPython

### DISCUSSION THREAD

**Discussion by cwerner** [Textualize/rich#1570] [Ideas]
Hi all. :wave:

I was wondering if it's possible to color-code dtypes of a pandas dataframe when pretty printing in the ipython REPL? 

For example: `str` in black, `int` in darkblue, `float` in dark green, `categorical` in purple, `object` (undefined) in gray?

I was thinking this might be helpful in data exploration using ipython - especially when the pandas read_csv magic to determine the most appropriate dtype fails again?

I'm very new to rich... Maybe this is already possible?

Cheers 🙏

**Comment by saschalalala** [comment]
I think, the highlighting of the dataframe is based on what the highlighter detects via regex:

<img width="1069" alt="grafik" src="https://user-images.githubusercontent.com/3439311/137374570-a644109b-3abd-4941-b0b0-5cc7c99f0dfe.png">

I am not sure if the renderer (print, ipython html, whatever) receives the required datatype information.
